anymaniax/orval
# fix(msw): correctly mock nested objects

## Status

**READY**

## Description

This fixes the issue with nested objects that are more than one level deep (in petstore, there is an example of Pet, which is one of Dog or Cat, and Dog is one of Labradoodle or Dachshund). There is a problem where only the last oneOf object gets the base object properties. This PR fixes this issue, more info in Issue #1210.

## Steps to Test or Reproduce

Before new code: go to tests/generated/swr/petstore/endpoints.ts, notice how `getListPetsResponseMock` contains:
```ts
faker.helpers.arrayElement([
      {
        breed: faker.helpers.arrayElement(['Labradoodle'] as const),
        cuteness: faker.number.int({ min: undefined, max: undefined }),
        ...overrideResponse,
        barksPerMinute: faker.helpers.arrayElement([
          faker.number.int({ min: undefined, max: undefined }),
          undefined,
        ]),
        type: faker.helpers.arrayElement(['dog'] as const),
        ...overrideResponse,
      },
      {
        breed: faker.helpers.arrayElement(['Dachshund'] as const),
        length: faker.number.int({ min: undefined, max: undefined }),
        ...overrideResponse,
        barksPerMinute: faker.helpers.arrayElement([
          faker.number.int({ min: undefined, max: undefined }),
          undefined,
        ]),
        type: faker.helpers.arrayElement(['dog'] as const),
        ...overrideResponse,
        '@id': faker.helpers.arrayElement([faker.word.sample(), undefined]),
        callingCode: faker.helpers.arrayElement([
          faker.helpers.arrayElement(['+33', '+420', '+33'] as const),
          undefined,
        ]),
        country: faker.helpers.arrayElement([
          faker.helpers.arrayElement([
            "People's Republic of China",
            'Uruguay',
          ] as const),
          undefined,
        ]),
        email: faker.helpers.arrayElement([faker.internet.email(), undefined]),
        id: faker.number.int({ min: undefined, max: undefined }),
        name: faker.word.sample(),
        tag: faker.helpers.arrayElement([faker.word.sample(), undefined]),
        ...overrideResponse,
      },
...
```

Notice that the first object is missing '@id', callingCode, country, email, id, name and tag.

With new code, do the same generation and check the same file. See that the missing properties are now present in the first object.
